src/App.js

```diff
@@ -1,5 +1,91 @@
+import { Console } from "@woowacourse/mission-utils";
+import {
+  CustomSeparatorError,
+  ValidateError,
+  ParseError,
+} from "./CustomErrors.js";
+import { CUSTOM_SEPARATOR_REG, DEGREE_REG } from "./constants.js";
+
 class App {
-  async run() {}
+  constructor() {
+    this.sum = 0;
+    this.separator = ",:";
+    this.userInput = "";
+  }
+
+  async run() {
+    await this.processUserInputAndStoreResult();
+    this.processCustomSeparator();
+    if (this.userInput.length > 0) {
+      const stringNumbers = this.getNumbers();
+      const numbers = this.parseNumbers(stringNumbers);
+      this.calculateSum(numbers);
+    }
+    this.printResult();
+  }
+
+  async processUserInputAndStoreResult() {
+    this.userInput = await Console.readLineAsync(
+      "덧셈할 문자열을 입력해 주세요.\n"
+    );
+  }
+
+  printResult() {
+    Console.print(`결과 : ${this.sum}`);
+  }
+
+  processCustomSeparator() {
+    const customSeparator = this.userInput.search(CUSTOM_SEPARATOR_REG);
+    if (customSeparator >= 0) {
+      this.checkCustomSeparatorError(this.userInput);
+      this.separator += this.userInput[2];
+      this.userInput = this.userInput.slice(5);
+    }
+  }
+
+  getNumbers() {
+    this.checkValidateError(this.userInput);
+    const separatorReg = new RegExp(`[${this.separator}]`, "g");
+    return this.userInput.split(separatorReg);
+  }
+
+  parseNumbers(stringNumbers) {
+    const parsedNumbers = stringNumbers.map((number) => {
+      const num = parseInt(number, 10);
+      this.checkParseError(num);
+      return num;
+    });
+    return parsedNumbers;
+  }
```

Comment:
```suggestion
  parseNumbers(stringNumbers) {
    return stringNumbers.map((number) => {
      const num = parseInt(number, 10);
      this.checkParseError(num);
      return num;
    });
  }
```

요렇게 축약할 수 있을 것 같아요! (물론 취향 차이인 것 같긴 합니다 ㅎㅎ 지금도 좋아용)

그리고 이 메서드가 하는 일이 사실 `stringNumbers`의 요소들을 숫자로 바꿔주고 그 과정에서 `validation`을 해주는 것 같은데 네이밍을 `parse`라고 하신 이유가 궁금합니다!  `convertStringsToNumbers`와 같은 이름은 어떨까요?